portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c

```diff
@@ -165,87 +165,150 @@ static u32 ulPHYIndex;
+            pxEMAC_PS = &( xEMACpsif.emacps );
+            memset( &xEMACpsif, '\0', sizeof( xEMACpsif ) );
 
-        if( xStatus != XST_SUCCESS )
-        {
-            FreeRTOS_printf( ( "xEMACInit: EmacPs Configuration Failed....\n" ) );
-        }
+            xStatus = XEmacPs_CfgInitialize( pxEMAC_PS, &mac_config, mac_config.BaseAddress );
+
+            if( xStatus != XST_SUCCESS )
+            {
+                FreeRTOS_printf( ( "xEMACInit: EmacPs Configuration Failed....\n" ) );
+                eEMACState = xEMAC_Fatal;
+                break;
+            }
 
 /* _HT_ : the use of jumbo frames has not been tested sufficiently yet. */
 
-        if( pxEMAC_PS->Version > 2 )
-        {
-            #if ( USE_JUMBO_FRAMES == 1 )
-                /* Enable jumbo frames for zynqmp */
-                XEmacPs_SetOptions( pxEMAC_PS, XEMACPS_JUMBO_ENABLE_OPTION );
-            #endif
-        }
+            if( pxEMAC_PS->Version > 2 )
+            {
+                #if ( USE_JUMBO_FRAMES == 1 )
+                    /* Enable jumbo frames for zynqmp */
+                    XEmacPs_SetOptions( pxEMAC_PS, XEMACPS_JUMBO_ENABLE_OPTION );
+                #endif
+            }
+
+            /* Initialize the mac and set the MAC address. */
+            XEmacPs_SetMacAddress( pxEMAC_PS, ( void * ) ipLOCAL_MAC_ADDRESS, 1 );
+
+            #if ( ipconfigUSE_LLMNR == 1 )
+                {
+                    /* Also add LLMNR multicast MAC address. */
+                    XEmacPs_SetMacAddress( pxEMAC_PS, ( void * ) xLLMNR_MACAddress, 2 );
+                }
+            #endif /* ipconfigUSE_LLMNR == 1 */
 
-        /* Initialize the mac and set the MAC address. */
-        XEmacPs_SetMacAddress( pxEMAC_PS, ( void * ) ipLOCAL_MAC_ADDRESS, 1 );
+            XEmacPs_SetMdioDivisor( pxEMAC_PS, MDC_DIV_224 );
+            ulPHYIndex = ulDetecPHY( pxEMAC_PS );
 
-        #if ( ipconfigUSE_LLMNR == 1 )
+            if( ulPHYIndex == ~0U )
             {
-                /* Also add LLMNR multicast MAC address. */
-                XEmacPs_SetMacAddress( pxEMAC_PS, ( void * ) xLLMNR_MACAddress, 2 );
+                FreeRTOS_printf( ( "xEMACInit: No valid PHY was found\n" ) );
+                eEMACState = xEMAC_Fatal;
+                break;
             }
-        #endif /* ipconfigUSE_LLMNR == 1 */
 
-        XEmacPs_SetMdioDivisor( pxEMAC_PS, MDC_DIV_224 );
-        ulPHYIndex = ulDetecPHY( pxEMAC_PS );
-        ulLinkSpeed = Phy_Setup_US( pxEMAC_PS, ulPHYIndex );
-        XEmacPs_SetOperatingSpeed( pxEMAC_PS, ulLinkSpeed );
+            eEMACState = xEMAC_SetupPHY;
 
-        /* Setting the operating speed of the MAC needs a delay. */
-        vTaskDelay( pdMS_TO_TICKS( 25UL ) );
+        /* Fall through. */
 
-        ulDMAReg = XEmacPs_ReadReg( pxEMAC_PS->Config.BaseAddress, XEMACPS_DMACR_OFFSET );
-        /* Enable 16-bytes AHB bursts */
-        ulDMAReg = ulDMAReg | XEMACPS_DMACR_INCR16_AHB_BURST;
+        case xEMAC_SetupPHY:
```

Comment:
If we are always going to fall through, what is the point of having this case instead of a comment?
Because all the breaks end with fatal error.